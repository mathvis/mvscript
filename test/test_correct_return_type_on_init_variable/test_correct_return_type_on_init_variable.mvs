func a() bool {}
let b: bool = a();
