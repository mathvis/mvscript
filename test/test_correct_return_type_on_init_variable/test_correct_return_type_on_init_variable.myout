Decl (FunctionDef (FunctionIdentifier "a") [] BoolT (Just (Block (FunctionBlock BoolT) [Expr (Return (Just (Type (Bool False))))])))
Decl (Variable (VarIdentifier "b") (Just BoolT) (Just (FunctionCall (FunctionIdentifier "a") [])))
