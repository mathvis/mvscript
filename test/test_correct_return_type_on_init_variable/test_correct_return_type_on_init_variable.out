Decl (FunctionDef (FunctionIdentifier "a") [] BoolT (Just (Block (FunctionBlock BoolT) [])))
Decl (Variable (VarIdentifier "b") (Just BoolT) (Just (FunctionCall (FunctionIdentifier "a") [])))
